backend/app/models/llm/request.py

```
@@ -0,0 +1,81 @@
+from typing import Any, Literal
+
+from sqlmodel import Field, SQLModel
+from pydantic import model_validator, HttpUrl
+
+
+class ConversationConfig(SQLModel):
+    id: str | None = Field(
+        default=None,
+        description=(
+            "Identifier for an existing conversation. "
+            "Used to retrieve the previous message context and continue the chat. "
+            "If not provided, a new conversation will be created."
+        ),
+    )
+    auto_create: bool = Field(
+        default=False,
+        description=(
+            "Only if True and no `id` is provided, a new conversation will be created automatically."
+        ),
+    )
+
+    @model_validator(mode="after")
+    def validate_conversation_logic(self):
```

could lead to confusing behavior for the API caller

what if they erroneously send a conversation id but actually intend to auto_create - in this case we are arbitrarily choosing to give priority to the conversation id and ignoring the auto_create request

probably raise validation error instead of silently modifying?

---

add a validation error